controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -137,3 +148,64 @@ func (r *K8ssandraClusterReconciler) reconcileMedusaConfigMap(
 	logger.Info("Medusa ConfigMap successfully reconciled")
 	return result.Continue()
 }
+
+func (r *K8ssandraClusterReconciler) reconcileMedusaStandalone(
+	ctx context.Context,
+	kc *api.K8ssandraCluster,
+	remoteClient client.Client,
+	desiredDeployment *appsv1.Deployment,
+	desiredService *corev1.Service,
+	logger logr.Logger,
+) result.ReconcileResult {
+
+	logger.Info("Reconciling Medusa standalone deployment on namespace : " + desiredDeployment.ObjectMeta.Namespace)
+	if kc.Spec.Medusa != nil {
+		deploymentKey := utils.GetKey(desiredDeployment)
+
+		logger := logger.WithValues("MedusaDeployment", deploymentKey)
+
+		// Compute a hash which will allow to compare desired and actual configMaps
+		annotations.AddHashAnnotation(desiredDeployment)
+
+		actualDeployment := &appsv1.Deployment{}
+		if err := remoteClient.Get(ctx, deploymentKey, actualDeployment); err != nil {
+			if errors.IsNotFound(err) {
+				if err := remoteClient.Create(ctx, desiredDeployment); err != nil {
+					logger.Error(err, "Failed to create Medusa deployment")
+					return result.Error(err)
+				}
+			}
+		}
+
+		actualDeployment = actualDeployment.DeepCopy()
+
+		if !annotations.CompareHashAnnotations(actualDeployment, desiredDeployment) {
+			logger.Info("Updating Medusa deployment on namespace " + actualDeployment.ObjectMeta.Namespace)
+			resourceVersion := actualDeployment.GetResourceVersion()
+			desiredDeployment.DeepCopyInto(actualDeployment)
+			actualDeployment.SetResourceVersion(resourceVersion)
+			if err := remoteClient.Update(ctx, actualDeployment); err != nil {
+				logger.Error(err, "Failed to update Medusa deployment resource")
+				return result.Error(err)
+			}
+			return result.RequeueSoon(r.DefaultDelay)
+		}
+
+		// Create a service for the medusa standalone deployment
```

Comment:
This logic does not use the `CompareHashAnnotations` logic typically used in reconciliation. I'm open to not using it because `CompareHashAnnotations` doesn't actually allow for real self healing (the hash annotation is not updated when a user manually modifies an object, so an object can drift without being detected).

Just wondering why we're using two different methodologies really, I don't think it is necessarily a problem.

Reply:
In the end I decided not to use the hash for the service. There's really nothing to configure there, so once it's created then we're fine. I'm reluctant to add more code to handle this, as it's not really necessary.

Reply:
Did we determine that changes to labels etc. would not cause an update of the service? As long as changes flowing from the CRs do not cause uncaught updates to the service I'm fine with not using a hash comparison.

Reply:
> As long as changes flowing from the CRs do not cause uncaught updates to the service I'm fine with not using a hash comparison.

The K8ssandraCluster name is all that could trigger a breaking change here and it's not possible, so we're safe 👍